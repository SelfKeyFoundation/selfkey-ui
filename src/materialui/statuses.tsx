import * as React from 'react';
import classNames from 'classnames';
import {
    withStyles,
    Typography,
    Grid,
    Theme,
    createStyles,
    Button,
} from '@material-ui/core';
import { 
    AttributeAlertLargeIcon,
    SimpleCheckIcon,
    SimpleDeniedIcon,
    SimpleHourglassIcon,
    error,
    success,
    typography,
    warning
} from '../theme/selfkey-dark-theme';

export type StatusProps = any;

const statusInfoStyle = (theme: Theme) =>
    createStyles({
        defaultStatus: {
            border: `1px solid ${success}`,
            borderRadius: '4px',
            boxSizing: 'border-box',
            padding: '25px 30px',
        },
        grow: {
            flexGrow: 1
        },
        statusIcon: {
            marginRight: '25px'
        },
        iconContainer: {
            marginRight: '25px',
            textAlign: 'center',
            width: '38px'
        },
        attribute: {
            height: '45px',
            width: '38px'
        },
        statusWrap: {
            '& .DocumentsRequired': {
                border: `1px solid ${warning}`,
            },
            '& .DocumentsSubmitted': {
                border: `1px solid ${typography}`
            },
            '& .Denied': {
                border: `1px solid ${error}`
            }
        },
    });

const StatusIcon = (status: StatusProps) => {
    if (status.status === 'DocumentsRequired') {
        return (
            <AttributeAlertLargeIcon />
        )
    } else if (status.status === 'DocumentsSubmitted') {
        return (
            <SimpleHourglassIcon />
        )
    } else if (status.status === 'Denied') {
        return (
            <SimpleDeniedIcon />
        )
    } else {
        return (
            <SimpleCheckIcon />
        )
    }
};

const StatusMessage = (status: StatusProps) => {
    if (status.status === 'DocumentsRequired') {
        return (
            <Typography variant='subtitle2' color='secondary'>
                Application started. Missing required documents.
            </Typography>
        )
    } else if (status.status === 'DocumentsSubmitted') {
        return (
            <Typography variant='subtitle2' color='secondary'>
                Application started. Documents submitted. Please check your email for further instructions.
            </Typography>
        )
    } else if (status.status === 'Denied') {
        return (
            <Typography variant='subtitle2' color='secondary'>
                Application denied.  Please check your email to for the rejection reason.
            </Typography>
        )
    } else {
        return (
            <Typography variant='subtitle2' color='secondary'>
                Application completed. Please check your email to recieve relevant documents and information.
            </Typography>
        )
    }
};

const StatusButton = (status: StatusProps) => {
    if (status.status === 'DocumentsRequired') {
        return (
            <Button variant='contained' size='large'>Add Documents</Button>
        )
    } else {
        return (
            <span></span>
        )
    }
};

export const StatusInfo = withStyles(statusInfoStyle)(({ classes, status }: StatusProps) => (
    <div className={classes.statusWrap}>
        <Grid item className={classNames(classes.defaultStatus, classes.pending, status)}>
            <Grid container direction='row' justify='space-between' alignItems='center' wrap='nowrap'>
                <Grid item className={classes.iconContainer}>
                    <StatusIcon status={status} className={classes.statusIcon} />
                </Grid>
                <Grid item className={classes.grow}>
                    <Typography variant='h2'>Status</Typography>
                    <StatusMessage status={status} />
                </Grid>
                <Grid item>
                    <StatusButton status={status} />
                </Grid>
            </Grid>
        </Grid>
    </div>
));
