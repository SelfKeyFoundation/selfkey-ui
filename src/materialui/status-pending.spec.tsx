import * as React from 'react';
import { create } from 'react-test-renderer';
import { StatusInfo } from './statuses';


describe('Status Pending', () => {
  it('should show Status Pending', () => {
    const component = create(
      <StatusInfo status="pending" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});





