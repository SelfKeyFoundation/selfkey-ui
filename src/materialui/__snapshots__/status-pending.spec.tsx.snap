// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Pending should show Status Pending 1`] = `
<div
  className="Component-statusWrap-6"
>
  <div
    className="MuiGrid-item-8 Component-defaultStatus-1 pending"
  >
    <div
      className="MuiGrid-container-7 MuiGrid-wrap-xs-nowrap-13 MuiGrid-align-items-xs-center-15 MuiGrid-justify-xs-space-between-26"
    >
      <div
        className="MuiGrid-item-8 Component-iconContainer-4"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-104 Component-root-0-2-1"
          focusable="false"
          role="presentation"
          viewBox="0 0 27 45"
        >
          <title>
            icon-simple-hourglass
          </title>
          <g
            fill="none"
            fill-rule="evenodd"
            id="ðŸ’»-ID-Dashboard-simle-hourglass-icon"
            stroke="none"
            stroke-width="1"
          >
            <g
              fill="#93B0C1"
              id="410.1-Marketplace-Applications-Statuses-simple-hourglass-icon"
              transform="translate(-202.000000, -502.000000)"
            >
              <g
                id="Content"
                transform="translate(150.000000, 264.000000)"
              >
                <g
                  id="Collapse-box-1"
                >
                  <g
                    id="Status-box-2"
                    transform="translate(15.000000, 213.000000)"
                  >
                    <path
                      d="M63.72,37.4 L63.72,28.4 C63.72,26.7431458 62.3768542,25.4 60.72,25.4 L40.72,25.4 C39.0631458,25.4 37.72,26.7431458 37.72,28.4 L37.72,37.4 C37.721724,38.2515717 38.0852731,39.0622861 38.72,39.63 L47.35,47.4 L38.72,55.17 C38.0852731,55.7377139 37.721724,56.5484283 37.72,57.4 L37.72,66.4 C37.72,68.0568542 39.0631458,69.4 40.72,69.4 L60.72,69.4 C62.3768542,69.4 63.72,68.0568542 63.72,66.4 L63.72,57.4 C63.718276,56.5484283 63.3547269,55.7377139 62.72,55.17 L54.09,47.4 L62.72,39.63 C63.3547269,39.0622861 63.718276,38.2515717 63.72,37.4 L63.72,37.4 Z M60.72,66.4 L40.72,66.4 L40.72,65.4 L60.72,65.4 L60.72,66.4 Z M60.72,62.4 L40.72,62.4 L40.72,57.4 L49.78,49.24 C50.3854601,49.4511014 51.0445399,49.4511014 51.65,49.24 L60.72,57.4 L60.72,62.4 Z M60.72,37.4 L51.66,45.56 C51.0545399,45.3488986 50.3954601,45.3488986 49.79,45.56 L40.72,37.4 L40.72,28.4 L60.72,28.4 L60.72,37.4 Z"
                      id="icon-simple-hourglass"
                    />
                  </g>
                </g>
              </g>
            </g>
          </g>
        </svg>
      </div>
      <div
        className="MuiGrid-item-8 Component-grow-2"
      >
        <h2
          className="MuiTypography-root-113 MuiTypography-h2-126"
        >
          Status
        </h2>
        <h6
          className="MuiTypography-root-113 MuiTypography-subtitle2-132 MuiTypography-colorSecondary-144"
        >
          Application started. Documents submitted. Please check your email for further instructions.
        </h6>
      </div>
      <div
        className="MuiGrid-item-8"
      >
        <span />
      </div>
    </div>
  </div>
</div>
`;
